src/main/scala/mesosphere/marathon/state/AppDefinition.scala

```diff
@@ -274,11 +274,12 @@ case class AppDefinition(
     )
   }
 
-  private val portIndicies: Range = container.flatMap(_.hostPorts).getOrElse(portNumbers).indices
+  private val portIndices: Range = container.flatMap(_.hostPorts.filter(_.nonEmpty)).getOrElse(portNumbers).indices
```

Comment:
if it was correctly defined as a Seq[], yes, but its not, its an Option[Seq[]] - so we're saying "use the container's host ports if they are defined and non-empty" - we can't serialize Option[Seq[]] to proto, so there's a major inconsistency here.